src/objective/rank_obj.cu

```diff
@@ -242,98 +242,98 @@ struct MAPLambdaWeightComputer {
  private:
-  dh::device_vector<bst_float> dpreds;   // Used to store sorted predictions
-  dh::device_vector<bst_float> dlabels;  // Used to store sorted labels
-  const bst_float *orig_dpreds;          // Original predictions - unsorted
-  const bst_float *orig_dlabels;         // Original labels - unsorted
+  dh::device_vector<bst_float> dpreds_;   // Used to store sorted predictions
+  dh::device_vector<bst_float> dlabels_;  // Used to store sorted labels
+  const bst_float *orig_dpreds_;          // Original predictions - unsorted
+  const bst_float *orig_dlabels_;         // Original labels - unsorted
 
-  dh::device_vector<int> dpos;           // Original position of the labels in the dataset
-  dh::device_vector<int> dlabels_count;  // Unique label count in CSR format
-  cudaStream_t stream{nullptr};
-  int begin_group_idx{-1};               // Begining index within the group
-  int device_id{-1};                     // GPU device ID
+  dh::device_vector<int> dpos_;           // Original position of the labels in the dataset
+  dh::device_vector<int> dlabels_count_;  // Unique label count in CSR format
+  cudaStream_t stream_{nullptr};
+  int begin_group_idx_{-1};               // Begining index within the group
+  int device_id_{-1};                     // GPU device ID
 
  public:
   SortedLabelList(int dev_id,
                   const bst_float *preds, const bst_float *labels,
                   int begin, int end)
-    : dpreds(preds + begin, preds + end),
-      dlabels(labels + begin, labels + end),
-      orig_dpreds(preds),
-      orig_dlabels(labels),
-      dpos(end - begin),
-      dlabels_count(end - begin + 1, 1),
-      begin_group_idx(begin),
-      device_id(dev_id) {
-    dh::safe_cuda(cudaStreamCreate(&stream));
-    thrust::sequence(thrust::cuda::par.on(stream), dpos.begin(), dpos.end());
-    dlabels_count[0] = 0;
+    : dpreds_(preds + begin, preds + end),
+      dlabels_(labels + begin, labels + end),
+      orig_dpreds_(preds),
+      orig_dlabels_(labels),
+      dpos_(end - begin),
+      dlabels_count_(end - begin + 1, 1),
+      begin_group_idx_(begin),
+      device_id_(dev_id) {
+    dh::safe_cuda(cudaStreamCreate(&stream_));
+    thrust::sequence(thrust::cuda::par.on(stream_), dpos_.begin(), dpos_.end());
+    dlabels_count_[0] = 0;
   }
 
   ~SortedLabelList() {
-    dh::safe_cuda(cudaSetDevice(device_id));
-    dh::safe_cuda(cudaStreamDestroy(stream));
+    dh::safe_cuda(cudaSetDevice(device_id_));
+    dh::safe_cuda(cudaStreamDestroy(stream_));
   }
 
   // Sort by predictions first and then sort the labels by predictions next
   void Sort() {
-    dh::device_vector<bst_float> cdpreds(dpreds);
+    dh::device_vector<bst_float> cdpreds(dpreds_);
     // Sort the predictions first and rearrange its positional indices
-    thrust::sort_by_key(thrust::cuda::par.on(stream),
-                        cdpreds.begin(), cdpreds.end(), dpos.begin(),
+    thrust::sort_by_key(thrust::cuda::par.on(stream_),
```

Comment:
Did you find some advantage to launching this in its own stream compared to the default stream? Just curious.

Reply:
n/a anymore.